Framework/Core/include/Framework/HistogramRegistry.h

```diff
@@ -482,57 +482,36 @@ struct HistFiller {
 class HistogramRegistry
 {
  public:
-  HistogramRegistry(char const* const name_, std::vector<HistogramSpec> histSpecs_ = {}, OutputObjHandlingPolicy policy_ = OutputObjHandlingPolicy::AnalysisObject, bool sortHistos_ = true, bool createRegistryDir_ = false)
-    : mName(name_), mPolicy(policy_), mRegistryKey(), mRegistryValue(), mSortHistos(sortHistos_), mCreateRegistryDir(createRegistryDir_)
+  HistogramRegistry(char const* const name, std::vector<HistogramSpec> histSpecs = {}, OutputObjHandlingPolicy policy = OutputObjHandlingPolicy::AnalysisObject, bool sortHistos = true, bool createRegistryDir = false)
+    : mName(name), mPolicy(policy), mRegistryKey(), mRegistryValue(), mSortHistos(sortHistos), mCreateRegistryDir(createRegistryDir)
   {
     mRegistryKey.fill(0u);
-    for (auto& histSpec : histSpecs_) {
+    for (auto& histSpec : histSpecs) {
       insert(histSpec);
     }
   }
 
-  void add(const HistogramSpec& histSpec_)
-  {
-    insert(histSpec_);
-  }
-
-  void add(char const* const name_, char const* const title_, const HistogramConfigSpec& histConfigSpec_, bool callSumw2_ = false)
-  {
-    insert({name_, title_, histConfigSpec_, callSumw2_});
-  }
-
-  void add(char const* const name_, char const* const title_, HistType histType_, std::vector<AxisSpec> axes_, bool callSumw2_ = false)
-  {
-    insert({name_, title_, {histType_, axes_}, callSumw2_});
-  }
+  // functions to add histograms to the registry
+  void add(const HistogramSpec& histSpec);
+  void add(char const* const name, char const* const title, const HistogramConfigSpec& histConfigSpec, bool callSumw2 = false);
+  void add(char const* const name, char const* const title, HistType histType, std::vector<AxisSpec> axes, bool callSumw2 = false);
+  void addClone(const std::string& source, const std::string& target);
 
-  // store a copy of an existing histogram (or group of histograms) under a different name
-  void addClone(const std::string& source_, const std::string& target_)
+  // function to query if name is already in use
+  bool contains(char const* const name)
```

Comment:
Could you make the implementation for this private?

Reply:
I mean move it in the .cxx....

Reply:
I was just hoping that if I leave it in the header it is possible that the compile_time_hash() will be actually evaluated at compile-time if a string literal is passed as an argument (just speculating, though).
But sure I can move it to the .cxx.

Reply:
I think it will not be compile time evaluated in any case. If you want that, I think you need to have the function take a struct with the hash and have that struct constexpr constructible by const char const*.

Reply:
alternatively you could have `contains(int hash)` in the cxx file, and have a simple thin `contains(char const* const)`

Reply:
okay thanks! then I will give this a try.

Reply:
does this mean that the 'fill()' function hashes are also not evaluated at compile-time? (Or is it there somehow different because those are template functions?)
I think for fill() it would be good to have this hashing of the name done only once (at least when called with a string literal).

Reply:
Yes, absolutely, it needs to be compile time...